src/transformers/generation_flax_utils.py

```diff
@@ -283,6 +280,41 @@ def generate(
             # prepare decoder_input_ids for generation
             input_ids = jnp.ones((input_ids.shape[0], 1), dtype="i4") * decoder_start_token_id
 
+        # Prepare `max_length` depending on other stopping criteria
+        # if `max_new_tokens` is passed, but not `max_length` -> set `max_length = max_new_tokens`
+        input_ids_seq_length = input_ids.shape[-1]
+        if max_length is not None:
+            warnings.warn(
+                "The `max_length` argument is deprecated and will be removed in v5. Use `max_new_tokens` instead.",
+                FutureWarning,
+            )
+        if max_length is None and max_new_tokens is not None:
+            max_length = max_new_tokens + input_ids_seq_length
+        elif max_length is not None and max_new_tokens is not None:
```

Comment:
It is a bit weird to me that the code uses the deprecated argument. The code should use `max_new_tokens` if possible, and then we set `max_new_tokens` when `max_length` is provided, along with issuing the deprecation warning.